File: frontend/src/types/report.ts
export interface ReportGenerateRequest {
  format?: string;
  include_evidence?: boolean;
  include_recommendations?: boolean;
  include_timeline?: boolean;
  include_resolution?: boolean;
}

export interface ReportGenerateResponse {
  report_id: string;
  incident_id: string;
  format: string;
  generation_status: string;
}

export interface ReportListItem {
  id: string;
  incident_id: string;
  incident_number: string;
  incident_title: string;
  format: string;
  version: number;
  generation_status: string;
  created_at: string;
  generated_by: string | null;
}

export interface ReportDetail {
  id: string;
  incident_id: string;
  format: string;
  version: number;
  generation_status: string;
  created_at: string;
  generated_by: string | null;
  download_url: string;
  content: Record<string, unknown> | null;
}

File: backend/app/schemas/report.py
"""Incident report API schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.common import PaginatedResponse


class ReportGenerateRequest(BaseModel):
    format: str = Field(default="json")
    include_evidence: bool = True
    include_recommendations: bool = True
    include_timeline: bool = True
    include_resolution: bool = True


class ReportGenerateResponse(BaseModel):
    report_id: UUID
    incident_id: UUID
    format: str
    generation_status: str


class ReportListItem(BaseModel):
    id: UUID
    incident_id: UUID
    incident_number: str
    incident_title: str
    format: str
    version: int
    generation_status: str
    created_at: datetime
    generated_by: UUID | None = None


class ReportDetailResponse(BaseModel):
    id: UUID
    incident_id: UUID
    format: str
    version: int
    generation_status: str
    created_at: datetime
    generated_by: UUID | None = None
    download_url: str
    content: dict[str, Any] | None = None


class ReportListResponse(PaginatedResponse[ReportListItem]):
    pass
